crates/full-node/sov-sequencer/src/preferred/nonce_buffer_task.rs

```diff
@@ -134,22 +135,13 @@ impl NonPersistedTxs {
         }).or(Some(tx_nonce_check));
     }
 
-    fn mark_inflight_execution_failed(&mut self, tx_nonce_check: u64) {
+    fn mark_inflight_execution_failed(&mut self) {
         if !self.has_in_flight {
             let msg = "Sequencer nonce buffer: non-persisted tracking: attempted to mark executed tx successful when has_in_flight is false!";
             tracing::error!(msg);
             debug_assert!(false, "{msg}"); // See comment in mark_inflight
         }
         self.has_in_flight = false;
```

Comment:
If we have two txs in flight concurrently (because the caller submitted the same nonce twice), will this be incorrect?

Reply:
Actually great spot, thank you! For the same reason that this check isn't done in mark_succeeded. I had overlooked the second case here. Removing